src/main/java/com/zebrunner/carina/webdriver/DriverHelper.java

```diff
@@ -496,6 +506,192 @@ public void addCookie(Cookie cookie) {
+        return getDriver().manage().getCookieNamed(name);
+    }
+
+    /**
+     * get the value of a cookie that was a URL-encoded Map.
+     *
+     * @param name the cookie being targeted
+     * @return decoded cookie value
+     */
+    @SuppressWarnings("unchecked")
+    public Map<String, String> getDecodedValueOfCookieNamed(String name) {
+        String cookieJsonString = null;
+        try {
+            cookieJsonString = URLDecoder.decode(getCookieNamed(name).getValue(), StandardCharsets.UTF_8);
+        } catch (Exception e) {
+            LOGGER.error("Error caught while decoding the cookie with the name: " + name, e);
+        }
+        return new Gson().fromJson(cookieJsonString, HashMap.class);
+    }
+
+    public Set<Cookie> getAllCookies() {
+        return getDriver().manage().getCookies();
+    }
+
+    /**
+     * Return a cookie String similar to document.cookie in the browser console
+     * Includes the cookie name and value, and doesn't include the path, domain, etc for each cookie
+     * @return String with all the cookie names and their values
+     */
+    public String getAllCookiesString() {
+        return getAllCookies().stream()
+                .map(cookie -> cookie.getName() + "=" + cookie.getValue() + ";")
+                .collect(Collectors.joining());
+    }
+
+    /**
+     * delete a specific cookie
+     *
+     * @param name the name of the cookie being deleted
+     */
+    public void deleteCookieNamed(String name) {
+        getDriver().manage().deleteCookieNamed(name);
+    }
+
+    /**
+     * delete all cookies in the driver session
+     */
+    public void deleteAllCookies() {
+        getDriver().manage().deleteAllCookies();
+    }
+
+    /**
+     * add an item to local storage
+     *
+     * @param name the name of the item to save
+     * @param value the value of the item to save
+     */
+    public void addToLocalStorage(String name, String value) {
+        ((JavascriptExecutor) getDriver()).executeScript(String.format("window.localStorage.setItem('%s','%s');", name, value));
+    }
+
+    /**
+     * add an item to local storage
+     *
+     * @param name the name of the item to save
+     * @param value the value of the item to save
+     */
+    public void addToLocalStorage(String name, Boolean value) {
+        addToLocalStorage(name, String.valueOf(value));
+    }
+
+    /**
+     * check to see if the item is present in local storage
+     *
+     * @param name the name of the item being stored
+     * @return boolean for if the item is present or not
+     */
+    public boolean isItemPresentInLocalStorage(String name) {
+        return ((JavascriptExecutor) getDriver()).executeScript(String.format("return window.localStorage.getItem('%s');", name)) != null;
+    }
+
+    /**
+     * gets the value of the stored item
+     *
+     * @param name the item that is stored
+     * @return value of the item
+     */
+    public String getValueFromLocalStorage(String name) {
```

Comment:
Change return type to Optional<String>. It should return Optional.empty it there are no value with such name in local storage

Reply:
Done.